### PR TITLE
fix: missing revisions and readOnly in documents meta

### PR DESCRIPTION
**Pull request by shumkov**
## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: packages/rs-dpp/src/data_contract/config/mod.rs [hunk 1]
```diff
@@ -48,107 +46,73 @@ impl DataContractConfig {
...
-    fn keeps_history(&self) -> bool {
+    fn keep_previous_contract_versions(&self) -> bool {
         match self {
-            DataContractConfig::V0(v0) => v0.keeps_history,
+            DataContractConfig::V0(v0) => v0.keep_previous_contract_versions,
         }
     }
 
-    fn documents_keep_history_contract_default(&self) -> bool {
+    fn document_revisions(&self) -> bool {
```
Comment (QuantumExplorer) [inline packages/rs-dpp/src/data_contract/config/mod.rs:70]: No, let's keep documents_keep_history_contract_default, other ones I can be okay with, this one I can not.

File: packages/rs-dpp/src/data_contract/config/mod.rs [hunk 2]
```diff
@@ -48,107 +46,73 @@ impl DataContractConfig {
...
         match self {
-            DataContractConfig::V0(v0) => v0.documents_keep_history_contract_default,
+            DataContractConfig::V0(v0) => v0.document_revisions_default,
```
Comment (QuantumExplorer) [inline packages/rs-dpp/src/data_contract/config/mod.rs:72]: No, let's keep documents_keep_history_contract_default

File: packages/rs-dpp/src/data_contract/config/mod.rs [hunk 3]
```diff
@@ -48,107 +46,73 @@ impl DataContractConfig {
...
         }
     }
 
-    fn documents_mutable_contract_default(&self) -> bool {
+    fn documents_read_only(&self) -> bool {
```
Comment (QuantumExplorer) [inline packages/rs-dpp/src/data_contract/config/mod.rs:76]: Please revert to documents_mutable_contract_default, remember this is just the default for the contract, not that all documents are mutable/!read_only

File: packages/rs-dpp/src/data_contract/config/v0/mod.rs [hunk 4]
```diff
@@ -1,81 +1,83 @@
...
 pub struct DataContractConfigV0 {
     /// Can the contract ever be deleted. If the contract is deleted, so should be all
     /// documents associated with it. TODO: There should also be a way to "stop" the contract -
     /// contract and documents are kept in the system, but no new documents can be added to it
-    pub can_be_deleted: bool,
+    pub allow_contract_deletion: bool,
     /// Is the contract mutable. Means that the document definitions can be changed or new
     /// document definitions can be added to the contract
-    pub readonly: bool,
+    pub allowe_contract_update: bool,
```
Comment (QuantumExplorer) [inline packages/rs-dpp/src/data_contract/config/v0/mod.rs:25]: allow_contract_update

**Review by QuantumExplorer — CHANGES_REQUESTED** [review]

File: packages/rs-dpp/src/data_contract/config/v0/mod.rs [hunk 5]
```diff
@@ -1,81 +1,83 @@
...
 impl std::default::Default for DataContractConfigV0 {
     fn default() -> Self {
         DataContractConfigV0 {
-            can_be_deleted: DEFAULT_CONTRACT_CAN_BE_DELETED,
-            readonly: !DEFAULT_CONTRACT_MUTABILITY,
-            keeps_history: DEFAULT_CONTRACT_KEEPS_HISTORY,
-            documents_keep_history_contract_default: DEFAULT_CONTRACT_DOCUMENTS_KEEPS_HISTORY,
-            documents_mutable_contract_default: DEFAULT_CONTRACT_DOCUMENT_MUTABILITY,
+            allow_contract_deletion: DEFAULT_CONTRACT_CAN_BE_DELETED,
+            allowe_contract_update: !DEFAULT_CONTRACT_MUTABILITY,
```
Comment (QuantumExplorer) [inline packages/rs-dpp/src/data_contract/config/v0/mod.rs:77]: allow_contract update is the opposite of readonly so it should be allow_contract_update: DEFAULT_CONTRACT_MUTABILITY,

File: packages/rs-dpp/src/data_contract/config/v0/mod.rs [hunk 6]
```diff
@@ -89,54 +91,3 @@ impl DataContractConfigV0 {
...
-    /// # Parameters
-    ///
-    /// * `contract`: BTreeMap representing the contract.
-    /// * `platform_version`: The platform version being used.
-    ///
-    /// # Returns
-    ///
-    /// * `Result<ContractConfig, ProtocolError>`: On success, a ContractConfig.
-    ///   On failure, a ProtocolError.
-    pub(super) fn get_contract_configuration_properties_v0(
```
Comment (QuantumExplorer) [inline packages/rs-dpp/src/data_contract/config/v0/mod.rs:112]: why was this removed?

File: packages/rs-dpp/src/data_contract/document_type/class_methods/try_from_schema/v0/mod.rs [hunk 7]
```diff
@@ -44,9 +44,9 @@ impl DocumentTypeV0 {
         data_contract_id: Identifier,
         name: &str,
         schema: Value,
-        schema_defs: Option<&BTreeMap<String, Value>>,
-        default_keeps_history: bool,
-        default_mutability: bool,
+        schema_defs: Option<&Value>,
+        default_revisions: bool,
```
Comment (QuantumExplorer) [inline packages/rs-dpp/src/data_contract/document_type/class_methods/try_from_schema/v0/mod.rs:48]: default_keeps_history

File: packages/rs-dpp/src/data_contract/document_type/class_methods/try_from_schema/v0/mod.rs [hunk 8]
```diff
@@ -44,9 +44,9 @@ impl DocumentTypeV0 {
...
+        default_read_only: bool,
```
Comment (QuantumExplorer) [inline packages/rs-dpp/src/data_contract/document_type/class_methods/try_from_schema/v0/mod.rs:49]: default_mutability

File: packages/rs-dpp/src/data_contract/config/v0/mod.rs (hunk 6 above)
Comment (shumkov) [reply packages/rs-dpp/src/data_contract/config/v0/mod.rs:112]: Because we don't use this. It was needed when we had manually defined data contracts with JSON schema. If you need to get a config from value manually you just need to use platform_value::from_value()

File: packages/rs-dpp/src/data_contract/document_type/class_methods/try_from_schema/v0/mod.rs (hunk 8 above)
Comment (shumkov) [reply packages/rs-dpp/src/data_contract/document_type/class_methods/try_from_schema/v0/mod.rs:49]: Ok, I can bring mutability back and keep readOnly in JSON Schema since it is a special keyword for that.

File: packages/rs-dpp/src/data_contract/v0/methods/schema.rs [hunk 9]
```diff
@@ -60,13 +60,13 @@ impl DataContractMethodsV0 for DataContractV0 {
             .collect()
     }
 
-    fn schema_defs(&self) -> Option<&BTreeMap<DefinitionName, Value>> {
+    fn schema_defs(&self) -> Option<&Value> {
```
Comment (QuantumExplorer) [inline packages/rs-dpp/src/data_contract/v0/methods/schema.rs:63]: This will change lookup time from 1 to N

File: packages/rs-dpp/src/document/document_factory/v0/mod.rs [hunk 10]
```diff
@@ -330,7 +330,7 @@ impl DocumentFactoryV0 {
         documents
             .into_iter()
             .map(|(document, document_type)| {
-                if document_type.documents_mutable() {
+                if document_type.documents_read_only() {
```
Comment (QuantumExplorer) [inline packages/rs-dpp/src/document/document_factory/v0/mod.rs:333]: You are doing the opposite...

Comment (QuantumExplorer) [reply packages/rs-dpp/src/document/document_factory/v0/mod.rs:333]: Why are we changing this to documents_read_only... there are so many places in the code where we might overlook something now.

File: packages/rs-dpp/src/data_contract/v0/methods/schema.rs (hunk 9 above)
Comment (shumkov) [reply packages/rs-dpp/src/data_contract/v0/methods/schema.rs:63]: Lookup happens only in try_from_schema. We convert it to a map as far as I remember. But for user it's more convenient to deal with value since he can do:
```
let def = platform_value!({
   "a": {
        "type": "string"
    }
})

let doc1 = platform_value!({
  "properties": {
      $ref: "#/$defs"
   }
})
```

Comment (QuantumExplorer) [reply packages/rs-dpp/src/data_contract/v0/methods/schema.rs:63]: I don't understand.
